src/main/java/seedu/address/logic/commands/listingcommands/ListingCommandsUtil.java

```diff
@@ -0,0 +1,89 @@
+package seedu.address.logic.commands.listingcommands;
+
+import static seedu.address.logic.commands.listingcommands.AddBuyersToListingCommand.MESSAGE_PERSON_NOT_BUYER;
+import static seedu.address.logic.commands.listingcommands.AddListingCommand.MESSAGE_NOT_BUYER;
+import static seedu.address.logic.commands.listingcommands.AddListingCommand.MESSAGE_NOT_SELLER;
+
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.listing.Address;
+import seedu.address.model.listing.Area;
+import seedu.address.model.listing.Listing;
+import seedu.address.model.listing.Price;
+import seedu.address.model.listing.Region;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Role;
+
+/**
+ * Utility class containing helper methods for various Listing-related commands.
+ */
+public class ListingCommandsUtil {
+
+    /**
+     * Throws an exception if the person is not a Buyer. The error message is formatted using
+     * {@link seedu.address.logic.commands.listingcommands.AddBuyersToListingCommand#MESSAGE_PERSON_NOT_BUYER}
+     *
+     * @param person the person to check
+     * @param role the role of the person
+     * @param buyerIndex the index of the buyer in the list
+     * @throws CommandException if the person is a seller
+     */
+    public static void personIsNotBuyer(Person person, Role role, Index buyerIndex) throws CommandException {
```

Comment:
Handle ambiguity between personIsNotBuyer() and checkIfBuyer()? Both throw an exception if the person is not a Buyer but they return a CommandException with different error messages.

Two ways to do so :

1. Merge into a single method: 
public void verifyBuyerStatus(Person person, String errorMessage) throws CommandException {
    if (!person.isBuyer()) {
        throw new CommandException(errorMessage);
    }
}

2. Keep Both Methods with a Shared Helper Method
private void verifyBuyerStatus(Person person, String errorMessage) throws CommandException {
    if (!person.isBuyer()) {
        throw new CommandException(errorMessage);
    }
}

public void personIsNotBuyer(Person person) throws CommandException {
    verifyBuyerStatus(person, "Error message for personIsNotBuyer");
}

public void checkIfBuyer(Person person) throws CommandException {
    verifyBuyerStatus(person, "Error message for checkIfBuyer");
}